pkg/share/manager/cs3/cs3.go

```diff
@@ -315,11 +315,31 @@ func (m *Manager) ListShares(ctx context.Context, filters []*collaboration.Filte
 	if !ok {
 		return nil, errtypes.UserRequired("error getting user from context")
 	}
+	var rID *provider.ResourceId
+	if len(filters) != 0 {
+		grouped := share.GroupFiltersByType(filters)
+		for _, f := range grouped {
+			for _, f := range f {
+				if f.GetResourceId() != nil {
+					rID = f.GetResourceId()
```

Comment:
Is it guaranteed that `grouped` contains maximal one filter where `f.ResourceID != nil`? Otherwise this will return random results..

Reply:
fixed by using a slice of rIDs